### ISSUE TITLE
Long network ready times since updating to 234.11 from 233.75

### ISSUE THREAD

**Issue by graysky2** [systemd/systemd#6500]
### Submission type

 - [X] Bug report

### systemd version the issue has been seen with

> systemd 234

### Used distribution

> Arch Linux

### In case of bug report: Expected behaviour you didn't see

>The network should be up and available after the network.target is reached.

### In case of bug report: Unexpected behaviour you saw

> Application (kodi) complains about network being unreachable and fails to load properly.

### In case of bug report: Steps to reproduce the problem

> Have requisite services enabled and boot the machine.  Downgrading to version 233.75 of systemd fixes this bug.  Updating to 234.11 causes it (confirmed with 5 reboots on each version).

### Details

After updating to systemd 234.11-1, booting a machine that runs kodi using the below service results in errors to that application due to an unavailable network.  From the kodi.log:
```
ERROR: Unable to open database: MyVideos92 [2003](Can't connect to MySQL server on '10.1.10.2' (101 "Network is unreachable"))
```

Kodi is started using the following systemd service which it has for years without issue on this hardware:
```
# cat /usr/lib/systemd/system/kodi.service
[Unit]
Description=Starts instance of Kodi using xinit
After=systemd-user-sessions.service network.target sound.target mysqld.service
Conflicts=getty@tty1.service

[Service]
User=kodi
Group=kodi
PAMName=login
TTYPath=/dev/tty1
ExecStart=/usr/bin/xinit /usr/bin/kodi-standalone -- :0 -nolisten tcp vt1
Restart=on-abort
StandardInput=tty

[Install]
WantedBy=multi-user.target
```
If I add a `ExecStartPre=/usr/bin/sleep 2s` to the above service, kodi works as expected, connecting without issue.  Does the service unit I am using need to be modified to work as expected with version 234.11?  Any suggestions are welcomed as the `ExecStartPre` line I mentioned seems a bit hacky.

**Comment by zdzichu** [comment]
If you want reachable network, you should have Requires=/After= network-online.target

**Comment by graysky2** [comment]
> If you want reachable network, you should have Requires=/After= network-online.target

Thank you for the quick reply.  I modified the unit as you recommended and it does work as expected, but doing so introduces a delay of approx. ~17 second that was not present under version 233.75 using the simple `network.target` as shown above.  This makes me wonder if some systemd internals have changed in this regard going from 233.75 to 234.11.

To further support the "something changed with version 234.11 hypothesis," using the original systemd unit adding with the `ExecStartPre=` line calling `/usr/bin/sleep 2s` does lead to the expected, error-free behavior.  Since an additional 2 sec seems to be all that is needed, there are an extra ~12 sec of time unaccounted for, no?

Under 234.11 using the modified `kodi.service` (ie the one with `Requires=network-online` and `After= ... network-online`:
```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @16.754s
└─multi-user.target @16.753s
  └─kodi.service @16.753s
    └─network-online.target @16.751s
      └─systemd-networkd-wait-online.service @175ms +16.576s
        └─systemd-networkd.service @154ms +20ms
          └─systemd-udevd.service @130ms +23ms
            └─systemd-tmpfiles-setup-dev.service @105ms +23ms
              └─kmod-static-nodes.service @94ms +7ms
                └─system.slice @85ms
                  └─-.slice @79ms
```

If I downgrade to 233.75 and use the original `kodi.service` I posted above without those two modifications:
```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @2.037s
└─multi-user.target @2.037s
  └─kodi.service @2.037s
    └─mariadb.service @983ms +1.052s
      └─network.target @980ms
        └─systemd-networkd.service @965ms +14ms
          └─systemd-udevd.service @136ms +29ms
            └─systemd-tmpfiles-setup-dev.service @113ms +20ms
              └─systemd-sysusers.service @96ms +16ms
                └─systemd-remount-fs.service @82ms +8ms
                  └─systemd-journald.socket @77ms
                    └─-.mount @73ms
                      └─system.slice @74ms
                        └─-.slice @73ms
```

**Comment by graysky2** [comment]
I believe that this is a bug with systemd.  If I stop using systemd-networkd and systemd-resolved and instead switch over to a static netctl profile, I am able to boot and start kodi without any modification to the existing service file.  It come up without issue
```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @4.853s
└─multi-user.target @4.852s
  └─kodi.service @4.852s
    └─mariadb.service @3.838s +1.013s
      └─network.target @3.830s
        └─netctl@static.service @698ms +3.131s
          └─basic.target @697ms
            └─sockets.target @697ms
              └─dbus.socket @697ms
                └─sysinit.target @697ms
                  └─systemd-timesyncd.service @591ms +105ms
                    └─systemd-tmpfiles-setup.service @582ms +7ms
                      └─local-fs.target @581ms
                        └─srv-nfs4-music.mount @575ms +5ms
                          └─mnt-media.mount @464ms +45ms
                            └─systemd-fsck@dev-disk-by\x2dlabel-media.service @345ms +117ms
                              └─dev-disk-by\x2dlabel-media.device @345ms
```

EDIT: further, if I modify `kodi.service` to use the `After=systemd-user-sessions.service network-online.target sound.target mysqld.service` and `Requires=network-online.target` booting using netctl, the results are nearly identical and kodi starts without issue.  Compare the to the 16+ sec delay when I am using systemd-networkd:

```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @4.878s
└─multi-user.target @4.878s
  └─kodi.service @4.877s
    └─mariadb.service @3.849s +1.027s
      └─network.target @3.847s
        └─netctl@static.service @724ms +3.122s
          └─basic.target @722ms
            └─sockets.target @722ms
              └─dbus.socket @722ms
                └─sysinit.target @722ms
                  └─systemd-timesyncd.service @602ms +119ms
                    └─systemd-tmpfiles-setup.service @594ms +5ms
                      └─local-fs.target @594ms
                        └─srv-nfs4-music.mount @588ms +5ms
                          └─mnt-media.mount @477ms +28ms
                            └─systemd-fsck@dev-disk-by\x2dlabel-media.service @346ms +129ms
                              └─dev-disk-by\x2dlabel-media.device @346ms
```

**Comment by graysky2** [comment]
@poettering - Has anything changed that would affect this behavior from 234.11 to 233.75?

**Comment by poettering** [comment]
Any chance you can bisect systemd/networkd and find out which commit might be the culprit? Otherwise I am not sure what to suggest.

**Comment by graysky2** [comment]
I can try

**Comment by graysky2** [comment]
OK... bisected to c1835a427fddac41dcc780cf733b53e01721e8f2

```
% git bisect good d60c527
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect bad c1edab7 
Bisecting: 416 revisions left to test after this (roughly 9 steps)
[2cfafe4d03a24d41b8c96ad06e10251daf6740af] Merge pull request #5942 from keszybz/timestamp-writing
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect good       
Bisecting: 208 revisions left to test after this (roughly 8 steps)
[a877b1d9a7d89f708d91a907fac6aa52970ed7a8] Merge pull request #6151 from speakinghedge/6150-LLDP-messages-containing-padding-are-dropped
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect good
Bisecting: 103 revisions left to test after this (roughly 7 steps)
[74d8f1c55b5aa46d2745fa4a74ae9fced6a3cab8] Merge pull request #6194 from keszybz/urandom-magic
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect bad 
Bisecting: 52 revisions left to test after this (roughly 6 steps)
[887a40521dd6f52f7a30be468c0c05be48424a8b] Merge pull request #6201 from poettering/bus-driver-creds
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect bad
Bisecting: 25 revisions left to test after this (roughly 5 steps)
[f52b982cc9504bf89b035d4b55ac038a61226210] meson: stop setting the linker (#6182)
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect good
Bisecting: 13 revisions left to test after this (roughly 4 steps)
[8a380e9603cd2c621a5f06aea73777b294fffb9f] Merge pull request #6203 from poettering/import-assert
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect bad 
Bisecting: 5 revisions left to test after this (roughly 3 steps)
[9017f5d88d5061487de53f00a1a8c0a83e41e760] systemd-mount: support device names and multiple arguments for umount (#6096)
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect bad
Bisecting: 2 revisions left to test after this (roughly 2 steps)
[45a9eac9a0c85c0c76b46388a530edd335ae44af] networkd: Allow DHCP servers to be re-configured on carrier gain
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect bad
Bisecting: 0 revisions left to test after this (roughly 1 step)
[c1835a427fddac41dcc780cf733b53e01721e8f2] networkd: Wait for link to get carrier before setting addresses
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect bad
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[410a7f15f0c6a96cd3b6a73f4a6465a5abfb9d0a] networkd: Clean up pool addresses on link down
facade@ease /scratch/a/systemd-git (git)-[master|bisect] % git bisect good
c1835a427fddac41dcc780cf733b53e01721e8f2 is the first bad commit
commit c1835a427fddac41dcc780cf733b53e01721e8f2
Author: Richard Maw <richard.maw@codethink.co.uk>
Date:   Wed Jun 7 16:30:46 2017 +0100

    networkd: Wait for link to get carrier before setting addresses
    
    For containers the link is effectively always up,
    but for virtual and physical machines networkd may have started
    before the link has gained carrier.
    
    Networkd will configure addresses when carrier is gained,
    but should also configure addresses if the link is already up.
    Without this patch the addresses are set unconditionally.
    
    Normally this isn't a problem since addresses are either fixed,
    set over DHCP, or is never without carrier.
    But for machines that gain carrier and are configured to select
    an address from the unallocated local address pool
    this causes them to pick an address from the pool twice.
    
    This change to skip address configuration when a link is added
    before it has a carrier fixes having multiple addresses assigned
    if the machine starts networkd before it has gained carrier
    and is configured with an address from the pool.

:040000 040000 c32d10205139effacc502cf271588496c5525742 7beb3fd579a6871173ad67a18e96b6798e0774bf M	src
```

**Comment by graysky2** [comment]
...should have mentioned that reverting that commit when compiling 234.11 restores the expected behavior using the original service unit + systemd-networkd.

**Comment by zdzichu** [comment]
So my understanding is that it worked before by chance. You did not have `After=network-online.target`, the network was started in parallel with Kodi. This was classic race-condition, only Kodi was slightly slower and network configuration was finishing sooner.
After the commit above, network configuration takes longer and loses race condition to Kodi.

Two take aways:
1) using `After=network-online.target` is proper way to express real dependencies of Kodi and should be in. (provided your MySQL at 10.1.10.2 is really remote; if it's local, you could use `localhost` as a name; if it is remote, it won't be reachable without network link anyway)
2) it is worth investigation why network link  takes 17 seconds to be up.

**Comment by graysky2** [comment]
@zdzichu - Thank you for the insights.  The old state of things could be the race condition you describe with kodi being slower, but that also establishes that the network doesn't take >15 seconds to come up so something is still broken.

Let's explore your 2nd bullet point and I agree with you that the proper `kodi.service` should contain:
```
After=systemd-user-sessions.service network-online.target sound.target mysqld.service
Requires=network-online.target
```

https://github.com/graysky2/kodi-standalone-service/blob/master/init/kodi.service

If I use the Arch Linux provided systemd packages (no reverts or modifications), which is currently 234.11-6, booting with your modified service in place using netctl (static IP) gives `network.target @4.282s`, kodi comes up fine.  The complete output is:
```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @5.286s
└─multi-user.target @5.286s
  └─kodi.service @5.285s
    └─mariadb.service @4.284s +1.000s
      └─network.target @4.282s
        └─netctl@static.service @963ms +3.318s
          └─basic.target @930ms
            └─sockets.target @930ms
              └─dbus.socket @930ms
                └─sysinit.target @930ms
                  └─systemd-timesyncd.service @832ms +97ms
                    └─systemd-tmpfiles-setup.service @818ms +8ms
                      └─local-fs.target @818ms
                        └─srv-nfs4-misc.mount @815ms +2ms
                          └─mnt-media.mount @684ms +45ms
                            └─systemd-fsck@dev-disk-by\x2dlabel-media.service @567ms +116ms
                              └─dev-disk-by\x2dlabel-media.device @566ms
```

When I simply disable netctl and enable systemd-networkd (static IP) and reboot, I get  `network-online.target @17.274s` and kodi comes up fine. The complete output is:
```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @17.277s
└─multi-user.target @17.276s
  └─kodi.service @17.276s
    └─network-online.target @17.274s
      └─systemd-networkd-wait-online.service @184ms +17.089s
        └─systemd-networkd.service @154ms +28ms
          └─systemd-udevd.service @125ms +28ms
            └─systemd-tmpfiles-setup-dev.service @105ms +18ms
              └─kmod-static-nodes.service @95ms +2ms
                └─system.slice @79ms
                  └─-.slice @77ms
```

A few key questions:
- If the race condition is in fact true, that establishes that the network does not take >15 seconds to come alive.  Why does systemd think it's not up for that time period?
- Why are the outputs of `systemd analyze blame` so different when the only change is netctl vs systemd-networkd?

**Comment by zdzichu** [comment]
@richardmaw-codethink, @ssahani any ideas?

**Comment by graysky2** [comment]
Since it's been a week or so I can update by stating the observation is not unique to a single machine.  Confirmed behavior (bug??) on 2 others as well.  Temporary workaround is to switch over to netctl but would like to get to the bottom of this.

**Comment by poettering** [comment]
@graysky2 hmm, so in those 16s that networkd needs to configure your link, do you actually have a carrier on the wire? what do networkd logs show in those 16s? anything interesting? if you set SYSTEMD_LOG_LEVEL=debug env var for networkd, does it show anything interesting in those 16s then?

**Comment by graysky2** [comment]
I created `/etc/systemd/system/unit.d/systemd-networkd.conf` to contain the following:
```
[Service]
Environment=SYSTEMD_LOG_LEVEL=debug
```

Here is all I get:
```
% journalctl -b -u systemd-networkd
-- Reboot --
Aug 31 15:11:31 myth2 systemd[1]: Starting Network Service...
Aug 31 15:11:31 myth2 systemd-networkd[244]: Enumeration completed
Aug 31 15:11:31 myth2 systemd[1]: Started Network Service.
Aug 31 15:11:32 myth2 systemd-networkd[244]: eth0: Renamed to enp2s0
Aug 31 15:11:32 myth2 systemd-networkd[244]: enp2s0: IPv6 successfully enabled
Aug 31 15:11:35 myth2 systemd-networkd[244]: enp2s0: Gained carrier
Aug 31 15:11:36 myth2 systemd-networkd[244]: enp2s0: Gained IPv6LL
Aug 31 15:11:48 myth2 systemd-networkd[244]: enp2s0: Configured
```

**Comment by graysky2** [comment]
@poettering - This is still present and is introducing a 2+ min delay in another system.  Would like to get the bottom on the cause and fix:
```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @2min 417ms
└─multi-user.target @2min 416ms
  └─kodi.service @2min 414ms
    └─network-online.target @2min 411ms
      └─systemd-resolved.service @1.535s +148ms
        └─basic.target @1.532s
          └─sockets.target @1.532s
            └─dbus.socket @1.532s
              └─sysinit.target @1.530s
                └─systemd-update-done.service @1.507s +22ms
                  └─ldconfig.service @1.375s +130ms
                    └─local-fs.target @1.369s
                      └─run-user-1001-gvfs.mount @6.067s
                        └─run-user-1001.mount @1.952s
                          └─local-fs-pre.target @260ms
                            └─systemd-tmpfiles-setup-dev.service @237ms +21ms
                              └─systemd-sysusers.service @209ms +26ms
                                └─systemd-remount-fs.service @179ms +22ms
                                  └─system.slice @156ms
                                    └─-.slice @153ms
```

**Comment by jfrankenau** [comment]
> I created /etc/systemd/system/unit.d/systemd-networkd.conf to contain the following:
> ```
> [Service]
> Environment=SYSTEMD_LOG_LEVEL=debug
> ```

@graysky2, this needs to be in `/etc/systemd/system/systemd-networkd.service.d/override.conf` to take effect.

**Comment by graysky2** [comment]
@jfrankenau @poettering  - I created that file and booted using systemd-networkd.

```
% systemd-analyze critical-chain
The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @17.123s
└─multi-user.target @17.123s
  └─kodi.service @17.122s
    └─network-online.target @17.120s
      └─systemd-networkd-wait-online.service @191ms +16.927s
        └─systemd-networkd.service @163ms +27ms
          └─systemd-udevd.service @130ms +32ms
            └─systemd-tmpfiles-setup-dev.service @111ms +17ms
              └─kmod-static-nodes.service @94ms +11ms
                └─systemd-journald.socket @91ms
                  └─system.slice @80ms
                    └─-.slice @79ms
```

```
% journalctl -b -u systemd-networkd
-- Logs begin at Sat 2017-09-23 08:15:57 EDT, end at Mon 2017-10-30 19:31:08 EDT. --
Oct 30 19:30:02 quartz systemd[1]: Starting Network Service...
Oct 30 19:30:02 quartz systemd-networkd[267]: Failed to connect to bus, trying again in 5s: No such file or directory
Oct 30 19:30:02 quartz systemd-networkd[267]: Failed to open configuration file '/etc/systemd/networkd.conf': No such file or directory
Oct 30 19:30:02 quartz systemd-networkd[267]: timestamp of '/etc/systemd/network' changed
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /etc/systemd/network/other, because it's not a regular file with suffix .netdev.
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /etc/systemd/network/10-dhcp.network, because it's not a regular file with suffix .netdev.
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /usr/lib/systemd/network/80-container-host0.network, because it's not a regular file with suffix .netdev.
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /usr/lib/systemd/network/80-container-ve.network, because it's not a regular file with suffix .netdev.
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /usr/lib/systemd/network/80-container-vz.network, because it's not a regular file with suffix .netdev.
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /usr/lib/systemd/network/99-default.link, because it's not a regular file with suffix .netdev.
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /etc/systemd/network/other, because it's not a regular file with suffix .network.
Oct 30 19:30:02 quartz systemd-networkd[267]: Ignoring /usr/lib/systemd/network/99-default.link, because it's not a regular file with suffix .network.
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: Flags change: +LOOPBACK +UP +LOWER_UP +RUNNING
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: Link 1 added
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: link pending udev initialization...
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: Saved original MTU: 0
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: Adding address: ::1/128 (valid forever)
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: Adding address: 127.0.0.1/8 (valid forever)
Oct 30 19:30:02 quartz systemd-networkd[267]: rtnl: received address with invalid family 129, ignoring.
Oct 30 19:30:02 quartz systemd-networkd[267]: rtnl: received address with invalid family 128, ignoring.
Oct 30 19:30:02 quartz systemd-networkd[267]: Enumeration completed
Oct 30 19:30:02 quartz systemd[1]: Started Network Service.
Oct 30 19:30:02 quartz systemd-networkd[267]: eth0: Flags change: +MULTICAST +BROADCAST
Oct 30 19:30:02 quartz systemd-networkd[267]: eth0: Link 2 added
Oct 30 19:30:02 quartz systemd-networkd[267]: eth0: link pending udev initialization...
Oct 30 19:30:02 quartz systemd-networkd[267]: eth0: Saved original MTU: 1500
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: udev initialized link
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: Link state is up-to-date
Oct 30 19:30:02 quartz systemd-networkd[267]: No virtualization found in DMI
Oct 30 19:30:02 quartz systemd-networkd[267]: No virtualization found in CPUID
Oct 30 19:30:02 quartz systemd-networkd[267]: Virtualization XEN not found, /proc/xen/capabilities does not exist
Oct 30 19:30:02 quartz systemd-networkd[267]: This platform does not support /proc/device-tree
Oct 30 19:30:02 quartz systemd-networkd[267]: No virtualization found in /proc/cpuinfo.
Oct 30 19:30:02 quartz systemd-networkd[267]: This platform does not support /proc/sysinfo
Oct 30 19:30:02 quartz systemd-networkd[267]: Found VM virtualization none
Oct 30 19:30:02 quartz systemd-networkd[267]: lo: Unmanaged
Oct 30 19:30:02 quartz systemd-networkd[267]: eth0: Renamed to enp2s0
Oct 30 19:30:02 quartz systemd-networkd[267]: enp2s0: udev initialized link
Oct 30 19:30:02 quartz systemd-networkd[267]: enp2s0: Link state is up-to-date
Oct 30 19:30:02 quartz systemd-networkd[267]: enp2s0: found matching network '/etc/systemd/network/10-dhcp.network'
Oct 30 19:30:02 quartz systemd-networkd[267]: enp2s0: Bringing link up
Oct 30 19:30:02 quartz systemd-networkd[267]: enp2s0: IPv6 successfully enabled
Oct 30 19:30:03 quartz systemd-networkd[267]: enp2s0: Flags change: +UP
Oct 30 19:30:03 quartz systemd-networkd[267]: LLDP: Started LLDP client
Oct 30 19:30:03 quartz systemd-networkd[267]: enp2s0: Started LLDP.
Oct 30 19:30:06 quartz systemd-networkd[267]: enp2s0: Flags change: +LOWER_UP +RUNNING
Oct 30 19:30:06 quartz systemd-networkd[267]: enp2s0: Gained carrier
Oct 30 19:30:06 quartz systemd-networkd[267]: enp2s0: Acquiring DHCPv4 lease
Oct 30 19:30:06 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): STARTED on ifindex 2
Oct 30 19:30:06 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): DISCOVER
Oct 30 19:30:07 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): DISCOVER
Oct 30 19:30:07 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): OFFER
Oct 30 19:30:07 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): REQUEST (requesting)
Oct 30 19:30:07 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): ACK
Oct 30 19:30:07 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): lease expires in 11h 59min 57s
Oct 30 19:30:07 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): T2 expires in 10h 29min 58s
Oct 30 19:30:07 quartz systemd-networkd[267]: DHCP CLIENT (0x11b540a6): T1 expires in 5h 59min 57s
Oct 30 19:30:07 quartz systemd-networkd[267]: enp2s0: DHCPv4 address 192.168.1.2/24 via 192.168.1.1
Oct 30 19:30:07 quartz systemd-networkd[267]: Setting transient hostname: 'quartz'
Oct 30 19:30:07 quartz systemd-networkd[267]: Not connected to system bus, ignoring transient hostname.
Oct 30 19:30:07 quartz systemd-networkd[267]: enp2s0: Updating address: 192.168.1.2/24 (valid for 12h)
Oct 30 19:30:07 quartz systemd-networkd[267]: enp2s0: Adding address: fe80::468a:5bff:fea0:547a/64 (valid forever)
Oct 30 19:30:07 quartz systemd-networkd[267]: enp2s0: Gained IPv6LL
Oct 30 19:30:07 quartz systemd-networkd[267]: enp2s0: Discovering IPv6 routers
Oct 30 19:30:07 quartz systemd-networkd[267]: NDISC: Started IPv6 Router Solicitation client
Oct 30 19:30:07 quartz systemd-networkd[267]: NDISC: Sent Router Solicitation, next solicitation in 4s
Oct 30 19:30:07 quartz systemd-networkd[267]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=1 reply_cookie=0 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.4 object=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=2 reply_cookie=0 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.4 object=n/a interface=n/a member=n/a cookie=3 reply_cookie=2 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RequestName cookie=3 reply_cookie=0 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.4 object=n/a interface=n/a member=n/a cookie=5 reply_cookie=3 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Setting transient hostname: 'quartz'
Oct 30 19:30:07 quartz systemd-networkd[267]: Sent message type=method_call sender=n/a destination=org.freedesktop.hostname1 object=/org/freedesktop/hostname1 interface=org.freedesktop.hostname1 member=SetHostname cookie=4 reply_cookie=0 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Got message type=signal sender=org.freedesktop.DBus destination=:1.4 object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=2 reply_cookie=0 error-name=n/a error-message=n/a
Oct 30 19:30:07 quartz systemd-networkd[267]: Got message type=signal sender=org.freedesktop.DBus destination=:1.4 object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=4 reply_cookie=0 error-name=n/a error-message=n/a
Oct 30 19:30:08 quartz systemd-networkd[267]: Got message type=method_return sender=:1.5 destination=:1.4 object=n/a interface=n/a member=n/a cookie=3 reply_cookie=4 error-name=n/a error-message=n/a
Oct 30 19:30:11 quartz systemd-networkd[267]: NDISC: Sent Router Solicitation, next solicitation in 8s
Oct 30 19:30:19 quartz systemd-networkd[267]: NDISC: No RA received before link confirmation timeout
Oct 30 19:30:19 quartz systemd-networkd[267]: NDISC: Invoking callback for 't'.
Oct 30 19:30:19 quartz systemd-networkd[267]: enp2s0: Configured
Oct 30 19:30:19 quartz systemd-networkd[267]: Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=5 reply_cookie=0 error-name=n/a error-message=n/a
Oct 30 19:30:19 quartz systemd-networkd[267]: NDISC: Sent Router Solicitation, next solicitation in 15s
Oct 30 19:30:35 quartz systemd-networkd[267]: NDISC: Sent Router Solicitation, next solicitation in 32s
Oct 30 19:31:08 quartz systemd-networkd[267]: NDISC: Sent Router Solicitation, next solicitation in 1min 2s
```

The complete journalctl output is here: https://gist.github.com/graysky2/7e229c1efd67cfd8f4000a4be8af16aa

**Comment by graysky2** [comment]
@jfrankenau @poettering - Does the debug output I provided shed any light on this issue?  It is still ongoing for me.

**Comment by jfrankenau** [comment]
Sorry, @graysky2, I've got no idea. I stumbled across your issue as I've got a similar problem where the critical point in the logs is the `LOWER_UP` flag for my network device which takes quite some time to get set. I probably need to open another issue.

**Comment by graysky2** [comment]
@poettering - Thoughts?

EDIT: I should add that I can ssh into the box (ie the network is up and functional) but systemd still waits to start the kodi.service.  Evidence:
```
% systemd-analyze critical-chain
Bootup is not yet finished. Please try again later.
```

**Comment by msilletti** [comment]
@poettering
I am facing the same problem @graysky2 described here on my kodi box sharing a mysql database. 40 second boot time.

**Comment by graysky2** [comment]
I believe further evidence that the issue might be with systemd's network-online target: 
* Replace the `network-online.target` with `network.target`
* Add a `sleep 3s` as a ExecStartPre 

These 2 changes result in error-free launches of the modified `kodi.service` in approx 5 second via systemd-analyze critical-chain vs 17 seconds when I use the network-online.target.

@poettering  - Thoughts?

For reference:
```
/usr/lib/systemd/system/kodi.service
[Unit]
Description=Starts instance of Kodi using xinit
After=systemd-user-sessions.service network.target sound.target mysqld.service
#After=systemd-user-sessions.service network-online.target sound.target mysqld.service
#Requires=network-online.target
Conflicts=getty@tty1.service
Requires=mysqld.service

[Service]
User=kodi
Group=kodi
PAMName=login
TTYPath=/dev/tty1
ExecStartPre=/bin/sleep 3
ExecStart=/usr/bin/xinit /usr/bin/kodi-standalone -- :0 -nolisten tcp vt1
Restart=on-abort
StandardInput=tty

[Install]
WantedBy=multi-user.target
```

**Comment by graysky2** [comment]
@poettering - Any thoughts?  The behavior persists with systemd v238.0.  Thanks!

**Comment by msilletti** [comment]
@poettering I am still facing this bug using Arch Linux systemd version 238.76-1.

**Comment by tomc603** [comment]
I suggest marking this as a duplicate of #8686. `systemd-networkd-wait-online` appears to fail starting due to RA not being received before a timeout. Waiting for the network to be marked online seems like the sane dependency to have in kodi's service file, but the network-online service appears broken.

**Comment by yuwata** [comment]
Please disable `IPv6AcceptRA=` if it is not necessary. Closing.
